deployment/helm/node-feature-discovery/templates/worker.yaml

```
@@ -13,6 +13,13 @@ metadata:
   {{- end }}
 spec:
   revisionHistoryLimit: {{ .Values.worker.revisionHistoryLimit }}
+  updateStrategy:
+    type: {{ ((.Values.worker).updateStrategy).type | default "RollingUpdate" }}
+    {{- if or (not ((.Values.worker).updateStrategy).type) (eq .Values.worker.updateStrategy.type "RollingUpdate") }}
+    rollingUpdate:
+      maxSurge: {{ (((.Values.worker).updateStrategy).rollingUpdate).maxSurge | default 1 }}
```

I think this doesn't work as intended, e.f. if you want to set maxSurge to 0. 

I suggest using something like this for the values:

```suggestion
      {{-with Values.worker.updateStrategy.rollingUpdate.maxSurge }}
      maxSurge: {{ . }}
      {{- end }}
```

Alternatively we could have a simple

```yaml
        {{- with .Values.worker.updateStrategy }}
          {{- toYaml . | nindent 2 }}
        {{- end}}
```